TacTok/models/term_encoder.py

```diff
@@ -130,10 +133,27 @@ def __init__(self, opts):
         super().__init__()
         self.opts = opts
         self.vocab = opts.vocab + nonterminals
-        self.input_gate = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, nonlinear=torch.sigmoid)
+        self.input_gate = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, opts,
+                                                nonlinear=torch.sigmoid)
         self.forget_gates = ForgetGates(opts.term_embedding_dim, self.vocab, opts)
-        self.output_gate = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, nonlinear=torch.sigmoid)
-        self.update_cell = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, nonlinear=torch.tanh)
+        self.output_gate = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, opts,
+                                                 nonlinear=torch.sigmoid)
+        self.update_cell = InputOutputUpdateGate(opts.term_embedding_dim, self.vocab, opts,
+                                                 nonlinear=torch.tanh)
+        occurances = pickle.load(open(opts.globals_file, 'rb'))
```

Comment:
I'd include some comments explaining this, since the files we load here may change depending on more options later (especially once we merge my latest code and Timothy's latest code), so we'll need a good sense of what's going on. What are each of these files? How do we change them? And so on. We will soon have separate vocabulary files for paths, globals, locals, and constructors, plus a single merged vocabulary.

Reply:
Added some comments here. It's not clear yet that we should use anything but the globals vocabulary though; adding in the locals seems to make things worse in the current experiments. I think switching away from just using globals is a separate pull-request/experiment.